modules/account/lib/Account.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace Jelix\Authentication\Account;
+
+class Account
+{
+    protected $name;
+
+    protected $email;
+
+    protected $status;
+
+    protected $create_date;
+
+    protected $provider;
+
+    protected $account_id;
+
+    public static $properties = array('name', 'email', 'status', 'create_date', 'provider', 'account_id');
+
+    public static $editableProperties = array('email');
+
+    /**
+     * @param \jDaoRecordBase $daoRecord The account dao entry
+     */
+    public function __construct($daoRecord)
+    {
+        foreach(self::$properties as $prop) {
```

Comment:
instead of a property  for each field, we could store $daoRecord and use it to retrieve values in other methods.